Mathlib/Data/LazyList/Basic.lean

```diff
@@ -101,8 +101,7 @@ instance : LawfulMonad LazyList := LawfulMonad.mk'
     | case1 =>
       simp only [Thunk.pure, LazyList.bind, LazyList.traverse, Id.pure_eq]
     | case2 _ _ ih =>
-      simp [LazyList.bind, LazyList.traverse, Seq.seq, Id.map_eq, append, Thunk.pure]
-      rw [← ih]
+      simp [LazyList.bind, LazyList.traverse, Seq.seq, Id.map_eq, append, Thunk.pure, ← ih]
```

Comment:
Is this non-terminal?

Reply:
It is flagged by the flexible linter, yes - as `simp` is a flexible tactic, but `rw` is rigid, and both act on `\goal`.